src/test/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -338,90 +332,306 @@ void testSkipMonitoredTripCheck(ShouldSkipTripTestCase testCase) throws Exceptio
+            false,
+            updatedTrip.journeyState.noLongerPossible,
+            "updated Trip should indicate it is still possible on at least one day of the week"
+        );
+    }
+
+    /**
+     * Tests whether an OTP request can be made and if the trip is properly updated after not being able to find a
+     * matching itinerary for all days of the week.
+     */
+    @Test
+    public void canMakeOTPRequestAndResolveNoLongerPossibleTrip() throws URISyntaxException,
+        CloneNotSupportedException {
+        // create a mock monitored trip and CheckMonitorTrip instance
+        CheckMonitoredTrip mockCheckMonitoredTrip = createCheckMonitoredTrip();
+        MonitoredTrip mockTrip = mockCheckMonitoredTrip.trip;
+        Persistence.monitoredTrips.create(mockTrip);
+
+        // create mock itinerary existence for trip for Mondays
+        mockTrip.itineraryExistence.monday = new ItineraryExistence.ItineraryExistenceResult();
+
+        // update trip to say that itinerary was not possible on Mondays as of the last check
+        mockTrip.itineraryExistence.monday.invalidDates.add("Mock date");
+
+        // set trip status to be upcoming
+        mockTrip.journeyState.tripStatus = TripStatus.TRIP_UPCOMING;
+
+        // update the target date to be an upcoming Monday within the CheckMonitoredTrip
+        mockCheckMonitoredTrip.targetDate = "2020-06-15";
+        mockCheckMonitoredTrip.targetZonedDateTime = noonMonday8June2020
+            .withDayOfMonth(15)
+            .withHour(8)
+            .withMinute(35);
+
+        // create an OTP mock to return
+        OtpResponse mockWeekdayResponse = otpDispatcherResponse.getResponse();
+        Itinerary mockMondayJune15Itinerary = mockWeekdayResponse.plan.itineraries.get(0);
+        // parse original itinerary date/time and then update mock itinerary to occur on Monday June 15, but at a time
+        // that does not match the previous itinerary
+        updateBaseItineraryTime(
+            mockMondayJune15Itinerary,
+            DateTimeUtils.makeOtpZonedDateTime(mockMondayJune15Itinerary.startTime)
+                .withDayOfMonth(15)
+                .withMinute(22) // this will cause an itinerary mismatch
+        );
+        OtpTestUtils.setupOtpMocks(List.of(mockWeekdayResponse));
+
+        // mock the current time to be 8:45am on Monday, June 15
+        DateTimeUtils.useFixedClockAt(
+            noonMonday8June2020
+                .withDayOfMonth(15)
+                .withHour(8)
+                .withMinute(45)
+        );
+
+        // execute makeOTPRequestAndUpdateMatchingItinerary method and verify the expected outcome
+        assertEquals(false, mockCheckMonitoredTrip.makeOTPRequestAndUpdateMatchingItinerary());
+
+        // fetch updated trip from persistence
+        MonitoredTrip updatedTrip = Persistence.monitoredTrips.getById(mockTrip.id);
+
+        // verify that status is active
+        assertEquals(
+            TripStatus.NO_LONGER_POSSIBLE,
```

Comment:
There should be an assertion to check that a notification is generated regarding the new trip status.

Reply:
Fixed in 1987213b6754ffa98f4748820b628be64f9443f4